pkg/repositories/transformers/node_execution.go

```diff
@@ -223,6 +234,31 @@ func UpdateNodeExecutionModel(
 	}
 	nodeExecutionModel.NodeExecutionUpdatedAt = &updatedAt
 	nodeExecutionModel.DynamicWorkflowRemoteClosureReference = dynamicWorkflowRemoteClosure
+
+	// In the case of dynamic nodes reporting DYNAMIC_RUNNING, the IsParent and IsDynamic bits will be set for this event.
+	// Update the node execution metadata accordingly.
+	if request.Event.IsParent || request.Event.IsDynamic {
+		var nodeExecutionMetadata admin.NodeExecutionMetaData
+		if len(nodeExecutionModel.NodeExecutionMetadata) > 0 {
+			if err := proto.Unmarshal(nodeExecutionModel.NodeExecutionMetadata, &nodeExecutionMetadata); err != nil {
+				return errors.NewFlyteAdminErrorf(codes.Internal,
+					"failed to unmarshal node execution metadata with error: %+v", err)
+			}
+		}
+		if request.Event.IsParent {
+			nodeExecutionMetadata.IsParentNode = true
+		}
+		if request.Event.IsDynamic {
+			nodeExecutionMetadata.IsDynamic = true
+		}
```

Comment:
```suggestion
		if request.Event.IsParent {
			nodeExecutionMetadata.IsParentNode = true
		}
		if request.Event.IsDynamic {
			nodeExecutionMetadata.IsDynamic = true
		}
```
```suggestion
		nodeExecutionMetadata.IsParentNode = request.Event.IsParent
                nodeExecutionMetadata.IsDynamic= request.Event.IsDynamic
```

Reply:
ah let me add a comment, not every event sends IsParent _and_ IsDynamic as an artifact of how propeller handles dynamic nodes. So i only want to explicitly set the fields, when they're set in the event itself.